test/test_torch.py

```diff
@@ -3623,9 +3623,9 @@ def test_exponential(self, device, dtype):
 
     @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=False, include_half=False))
     def test_cov(self, device, dtype):
-        shapes = [(1,), (10,),
-                  (1, 1), (10, 1), (50, 5),
-                  (1, 20), (5, 20)]
+        shapes = [(1,), (4,), (10,)] # single dim inps
+        nums = [1, 2, 5, 10, 20, 50]
+        shapes = shapes + list(product(nums, nums)) # concat 2 dim inps
```

Comment:
@mruberry was this how you wanted me to use products?

Reply:
I was thinking more like:

```
biases = (True, False)
rowvars = (True, False)
ddofs = (None, 0, 1)
for shape, rowvar, bias, ddof, fweight, aweight in product(shapes, rowvars, biases, ddofs, fweights, aweights):
  ...
```

Another option would just be to create test cases by hand like this:

```
test_cases = (
  (shape, rowvar, bias, ddof, fweights, aweights),
  ...
)

for test_case in test_cases:
  shape, rowvar, bias, ddof, fweights, aweights = test_case
  ...
```

Either of these approaches should clarify what's being tested and simplify the test.